### PR TITLE
Request TransportURL when deploying Neutron

### PR DESCRIPTION
**Pull request by karelyatin**
`transport_url` is configured in neutron.conf from the secret provided by TransportURL CR.

This requires rabbitmq CR to be running.

Signed-off-by: Yatin Karel <ykarel@redhat.com>

### CODE REVIEW

**Review by slawqo — APPROVED** [review]
/lgtm for me but conflicts needs to be resolved first

**Comment by karelyatin** [comment]
> /lgtm for me but conflicts needs to be resolved first

Done

File: pkg/neutronapi/initcontainer.go [hunk 1]
```diff
@@ -29,68 +30,85 @@ func GetInitContainer(init InitContainer) []corev1.Container {
...
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: "rabbitmq-transport-url-neutron-neutron-transport",
+					},
+					Key: "transport_url",
 				},
-				{
-					Name:  "Database",
-					Value: init.Database,
+			},
```
Comment (abays) [inline pkg/neutronapi/initcontainer.go:43]: It looks like this is duplicated.  Lines 88-101 seem to handle whether `TransportURL` should be included in the env vars.

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**